air/src/proof/mod.rs

```diff
@@ -204,58 +205,129 @@ fn get_conjectured_security(
     cmp::min(cmp::min(field_security, query_security) - 1, collision_resistance)
 }
 
-#[cfg(feature = "std")]
 /// Estimates proven security level for the specified proof parameters.
+#[cfg(feature = "std")]
 fn get_proven_security(
     options: &ProofOptions,
     base_field_bits: u32,
     lde_domain_size: u64,
     trace_domain_size: u64,
     collision_resistance: u32,
 ) -> u32 {
-    let extension_field_bits = (base_field_bits * options.field_extension().degree()) as f64;
+    let m_min: usize = 3;
+    let m_max = compute_upper_m(trace_domain_size);
+
+    // We cap the range to 1000 as the optimal m value will be in the lower range of [m_min, m_max]
+    // since increasing m too much will lead to a deterioration in the FRI commit soundness making
+    // any benefit gained in the FRI query soundess mute.
+    let m_max = cmp::min(m_max as u64, PROXIMITY_PARAMETER_UPPER);
```

Comment:
nit: I would probably move this into `compute_upper_m` function.

Reply:
Done!